### PR TITLE
Add melody metric for result

### LINKED ISSUE #51: [Metric for test] How melody is same as the input vocal?
Extract F0 and compare it?
We have to find the metric for ability to keep "melody" of Songsuro model.

### PR DESCRIPTION
**Pull request by minsing-jin**
The melody of the generated song by Songsuro have to same as original melody

close #51 

image

### CODE REVIEW

**Comment by minsing-jin** [comment]
Review
image

File: songsuro/evaluation/melody_evaluator.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import torchcrepe
+from sklearn.metrics import f1_score
+from songsuro.preprocess import load_audio, make_spectrogram, make_mel_spectrogram
+
+
+class MelodyLoss:
+	def __init__(self, wav_gt_path=None, wav_gen_path=None):
+		if wav_gt_path is not None and wav_gen_path is not None:
+			self.wav_gen = load_audio(wav_gen_path)
+			self.wav_gt = load_audio(wav_gt_path)
```
Comment (vkehfdl1) [inline songsuro/evaluation/melody_evaluator.py:11]: Why load audio again?
In validation and test process we can get ground truth mel-spectorgram and generated audio waveform as torch tensor. 
If we save it to wav file and load it again it will be super inefficient.

**Review by vkehfdl1 — CHANGES_REQUESTED** [review]

**Comment by minsing-jin** [comment]
I bring melody metric in [CARGAN paper github implementation.](https://github.com/descriptinc/cargan/blob/master/cargan/evaluate/objective/metrics.py)
image

File: songsuro/evaluation/metrics.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+###############################################################################
+# Pitch metrics
+###############################################################################
+
+# TODO: How to use PitchEvaluator in the evaluation pipeline? Find a way to integrate it with the evaluation loop. or not loop.
+# TODO: I don't know why this is in the loop.
+
+
+def compute_spectrogram_mae(
```
Comment (bwook00) [inline songsuro/evaluation/metrics.py:18]: How about use 'melspectrogram' than 'spectrogram'?

**Review by bwook00 — CHANGES_REQUESTED** [review]

**Review by vkehfdl1 — CHANGES_REQUESTED** [review]

**Review by vkehfdl1 — APPROVED** [review]
LGTM
